### PR TITLE
fix(ordering): improve backend ordering and use Map for frontend

### PR DESCRIPTION
**Pull request by ixio**
This should keep our ordering upgrade and repair campaign edition (as well as keep all the improvements to creation).

Well no the only thing that isn't kept is that initially your annotators list is the backend order (alphabetical order) however if you add some and then remove them : they are shown in the list in the ordered they have been removed. I believe the gain in code maintainability outweighs this "functionality" : the most important thing is that the initial list is in alphabetical order.

This PR can also be reviewed by doing a git diff on the changed files with commit "f1bac2f - feat(backend) : add created_at to dataset and campaign (#131) (Qouagga 2023-11-20 14:39:00)" that was just before the commit that added the original flawed ordering code. For example `git diff f1bac2f -- frontend/src/CreateAnnotationCampaign.js`.

### CODE REVIEW

File: frontend/src/utils.js [hunk 1]
```diff
@@ -48,19 +39,3 @@ export function getTagColor(tags: Map<string, string>, tag: string): string {
...
-  for (const key in objects) {
-    if (objects[key].id === searchValue) {
-      return key
-    }
-  }
-}
-
-export function findObjetById(objects, searchValue) {
-  for (const key in objects) {
-    if (objects[key].id === searchValue) {
```
Comment (ElodieENSTA) [inline frontend/src/utils.js:62]: Are `key` and `id` property the same ?
Is it the reason why this function is not needed anymore ?

**Review by ElodieENSTA — COMMENTED** [review]
Why did you need to change all objects to Map ?
A single array of the data isn't enough ?

File: frontend/src/utils.js (hunk 1 above)
Comment (ixio) [reply frontend/src/utils.js:62]: See the removed comment on top of `frontend/src/CreateAnnotationCampaign.js` for more context, I feel the code is more simple and maintainable if `key == id` which is the code I originally created with Objects. However we needed to keep the order given by backend, in order to keep working with Objects Dewi was forced to use such aux functions. However I find it better to use Maps to have both order and key == id so these functions are no longer needed.

File: frontend/src/CreateAnnotationCampaign.js [hunk 2]
```diff
@@ -192,20 +184,20 @@ class CreateAnnotationCampaign extends Component<CACProps, CACState> {
...
     new_ac_annotation_mode: 1,
-    dataset_choices: {},
-    spectro_choices: {},
-    confidence_indicator_set_choices: {},
-    annotation_set_choices: {},
-    annotator_choices: {},
+    dataset_choices: new Map(),
+    spectro_choices: new Map(),
+    confidence_indicator_set_choices: new Map(),
+    annotation_set_choices: new Map(),
```
[the remark this answers is not in the document]
Comment (ixio) [reply frontend/src/CreateAnnotationCampaign.js:199]: @rvovard how do you feel about this ? Should I update flow types or are you just going to change everything once you move to Typescript ? What would be best for you ?

If I change the following : 
```javascript
type ShowAnnotationSetProps = {
  annotation_sets: {
    [?number]: annotation_set_type
  },
  onChange: (event: SyntheticEvent<HTMLInputElement>) => void
};
```
What should I change it to ? `annotation_sets: Map`, `annotation_sets: new Map()`, `annotation_sets: Map(number: annotation_set_type)` ? I'm not sure what the syntax is for type annotation of Maps.

[the remark this answers is not in the document]
Comment (ElodieENSTA) [reply frontend/src/CreateAnnotationCampaign.js:199]: Syntax is 
```typescript
type ShowAnnotationSetProps = {
  annotation_sets: Map<number, annotation_set_type>;
  onChange: (event: SyntheticEvent<HTMLInputElement>) => void;
};
```

[the remark this answers is not in the document]
Comment (rvovard) [reply frontend/src/CreateAnnotationCampaign.js:195]: I'll try to answer to questions:

- The ListChooser component will need refactoring in the future, but Erwan it was useful when you created it.
- Elodie gave correct advice about typings. I won't change everything when moving to TypeScript, just try to set the good types and fix when it doesn't compile. Actually, this PR will help me a lot, there was many typing problems with the code you change here.

**Review by ElodieENSTA — APPROVED** [review]

**Pull request merged**